Double scrollbars

andrew-secret opened the pull request:
### This PR closes the following issue:

closes https://github.com/owncast/owncast/issues/2499

### Changes made:

- hide footer on mobile
- fix header height by removing `flex-basis: content;` CSS property and remove stickyness (`position: sticky;`)
- remove javascript height calculation for `lowerSectionMobile`
- apply `overflow: hidden;` to main layout body

andrew-secret committed:
hide footer on mobile to avoid double scrollbars

andrew-secret committed:
fix double scrollbars on desktop by removing scroll behaviour from main section

andrew-secret committed:
add scrollbar styles to body and html

gabek commented:
Oh weird. I'm testing on a Chrome-based browser and it seems to be scrolling oddly, like the entire page, including the header is scrolling up. But from your demo videos it doesn't look like you're having that problem.

![Peek 2022-12-26 11-07](https://user-images.githubusercontent.com/414923/209577318-badd988a-177e-4058-af2a-0c3f70d8174a.gif)

andrew-secret commented:
That's interesting. Maybe I have to setup some livestream in order to test it properly. But thank you for testing it. I'll try to reproduce your case.

You mentioned that I can setup OBS and connect it to my local app

gabek commented:
Yeah for sure. There's some documentation [here](https://owncast.online/docs/broadcasting/) about streaming software as well. And if you install ffmpeg (looks like you're [on a mac](https://phoenixnap.com/kb/ffmpeg-mac) so probably through homebrew) you can use the test script that we were talking about yesterday.

andrew-secret commented:
Oh, and I have some different behavior. It depends on the browser, I guess.

But do you have this behavior in the current branch as well?

https://user-images.githubusercontent.com/13103896/209578960-2872e3ed-5dc9-4a7b-8d3f-6c8d13b232ec.mov

gabek commented:
I'm not able to reproduce that scrolling issue in the `webv2` branch. And you're right, it's only happening when a stream is happening. It's fine when it's not. But it seems to be fine for you both ways!

![Peek 2022-12-26 11-46](https://user-images.githubusercontent.com/414923/209579445-a32602db-e69b-40e3-9a5e-7ceaa704e88c.gif)

andrew-secret commented:
I think I know the issue. Seems like we need to keep `overflow-y: scroll` on the content element. I'll push that change and maybe then you can have a look at it again. Sorry for this back and forth 😆

andrew-secret committed:
add vertical scrolling back to main section

gabek commented:
I'm not sure if you were ready for me to take a look again or not, but I saw you push up another commit so I checked it out. It seems like something with the height is still going on as the entire content (including chat) can scroll vertically together.

Sorry I keep finding things! 

![Peek 2022-12-26 16-00](https://user-images.githubusercontent.com/414923/209589321-790f6fab-6c31-4b11-9dd0-1b61897225fd.gif)

andrew-secret commented:
> I'm not sure if you were ready for me to take a look again or not, but I saw you push up another commit so I checked it out. It seems like something with the height is still going on as the entire content (including chat) can scroll vertically together.
> 
> Sorry I keep finding things!
> 
> ![Peek 2022-12-26 16-00](https://user-images.githubusercontent.com/414923/209589321-790f6fab-6c31-4b11-9dd0-1b61897225fd.gif) [ ![Peek 2022-12-26 16-00](https://user-images.githubusercontent.com/414923/209589321-790f6fab-6c31-4b11-9dd0-1b61897225fd.gif) ](https://user-images.githubusercontent.com/414923/209589321-790f6fab-6c31-4b11-9dd0-1b61897225fd.gif) [ ](https://user-images.githubusercontent.com/414923/209589321-790f6fab-6c31-4b11-9dd0-1b61897225fd.gif)

It's good that you find out all those issues 🤩
I'll try to deep dive into it today/trmw. It seems like we have to take a closer look at how we handle scrolling in the app in general. Because I already saw some other issue (e.g. Chat overflows) that might be related to it.

andrew-secret committed:
prevent header from collapsing and messing around with scrolling

andrew-secret committed:
prevent body from scrolling for main layout

andrew-secret committed:
move tablet breakpoint into breakpoint mixin

andrew-secret committed:
preffify code

andrew-secret committed:
remove js height calc for lower section on mobile and use css

andrew-secret committed:
enable scrolling within a tab on mobile

andrew-secret commented:
I hope this PR also closes https://github.com/owncast/owncast/issues/2503

gabek commented:
Sweet! It looks to be functioning properly :100:. The only weird artifact is a small margin on the far right. I think it's an unused scrollbar since if I set `overflow: hidden` on `Content_mainSection`.

![image](https://user-images.githubusercontent.com/414923/209866262-9ac5306c-fa4d-4b3e-8ec0-1b6348f3124c.png)

andrew-secret committed:
remove scrollbar from main section on mobile

andrew-secret commented:
Cool! Scrollbar should be gone on mobile now :)

andrew-secret committed:
adjust header height on mobile

andrew-secret commented:
Ok, my QA-self discovered one small thing. I adjusted the header height on mobile 😆

![image](https://user-images.githubusercontent.com/13103896/209874987-a2c84485-92fb-483e-9b0a-304197b7dcee.png)

--

and I removed the nested styling in `Header.module.scss` because it is not needed there. I hope this is okay for you?

gabek commented:
This all looks great! I'll do some additional testing on more devices and if anything comes up I'll file specifics. Thank you for working so much on this one, I know it was annoying!

gabek commented on web/components/layouts/Main.tsx:
```diff
@@ -21,6 +21,12 @@ import { ServerRenderedHydration } from '../ServerRendered/ServerRenderedHydrati
 
 import { Theme } from '../theme/Theme';
 
+const globalStyle = `
```
Since this is a static style, is there any way to put it in a CSS module like all the rest of the styles?  That `customStyles` is different because it comes from the API.

andrew-secret replied:
This might be a bit tricky. A CSS module is not allowing global selectors. And if I try to add just some styles via `Main.scss` NextJS is complaining about it that I can't put global styles there.

I can take a closer look at it tmrw maybe I'll find something 🙂

andrew-secret replied:
I've added the `useLockBodyScroll` hook from `react-use` to it. It works.

After reading through some articles and stackoverflow, this seemed to be an appropriate solution. Since NextJS and CSS Modules really don't want to apply global styles in a module. Everybody was pointing either to the global Style solution in the head or using:

https://nextjs.org/blog/styling-next-with-styled-jsx

gabek commented:
I did add just one comment just as far as CSS organization.

andrew-secret committed:
add react use to web

andrew-secret committed:
lock body scroll for main layout

gabek commented:
Very cool! Seems to be working great. Going to merge this in. Thank you so much for finding solutions to all of these quirks!
